tests/framework/stats/function.py

```diff
@@ -163,8 +163,9 @@ def __call__(self, result: List) -> Any:
         length = len(result)
         result.sort()
         idx = length * self.k / 100
-        if idx is not int(idx):
-            return (result[int(idx)] + result[(int(idx) + 1)]) / 2
+        if idx != float(int(idx)):
```

Comment:
The `float` conversion is not necessary. 

nit: If find this logic more clear when using the `float` `is_integer` dedicated method. Multiple occurrences of `int(idx)` can be avoided as well by binding its result to a variable.